SimPPS/RPDigiProducer/plugins/RPGaussianTailNoiseAdder.cc

```diff
@@ -0,0 +1,46 @@
+#include "SimPPS/RPDigiProducer/plugins/RPGaussianTailNoiseAdder.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include <iostream>
+#include "TMath.h"
+#include "TRandom.h"
+#include "CLHEP/Random/RandGauss.h"
+
+using namespace std;
+
+RPGaussianTailNoiseAdder::RPGaussianTailNoiseAdder(int numStrips,
+                                                   double theNoiseInElectrons,
+                                                   double theStripThresholdInE,
+                                                   int verbosity)
+    : numStrips_(numStrips), theNoiseInElectrons(theNoiseInElectrons), theStripThresholdInE(theStripThresholdInE) {
+  verbosity_ = verbosity;
+  strips_above_threshold_prob_ = TMath::Erfc(theStripThresholdInE / sqrt(2.0) / theNoiseInElectrons) / 2;
+}
+
+simromanpot::strip_charge_map RPGaussianTailNoiseAdder::addNoise(const simromanpot::strip_charge_map &theSignal) {
+  simromanpot::strip_charge_map the_strip_charge_map;
+
+  // noise on strips with signal:
+  for (simromanpot::strip_charge_map::const_iterator i = theSignal.begin(); i != theSignal.end(); ++i) {
+    double noise = CLHEP::RandGauss::shoot(0.0, theNoiseInElectrons);
+    the_strip_charge_map[i->first] = i->second + noise;
+    if (verbosity_)
+      edm::LogInfo("RPDigiProducer") << "noise added to signal strips: " << noise << "\n";
+  }
+
+  // noise on the other strips
+  int strips_no_above_threshold = gRandom->Binomial(numStrips_, strips_above_threshold_prob_);
```

Comment:
@mundim , how gRandom is initialized?

Reply:
@mundim , there are equivalent CLHEP methods. Not necessary to use TRandom at all.

Reply:
Changed to the CLHEP equivalent. It was also replaced the TMath::Erfc by the std::erfc as advised by Ianna at the time of the PPS Geometry PR.